plugin/keyring_vault/vault_io.cc

```diff
@@ -75,7 +76,16 @@ bool Vault_io::get_serialized_object(ISerialized_object **serialized_object) {
   }
 
   std::unique_ptr<Vault_keys_list> keys(new Vault_keys_list());
-  if (vault_parser->parse_keys(json_response, keys.get())) {
+  auto key_callback = [&keys, this](std::unique_ptr<Vault_key> key) {
+    // With Vault API v2 server returns keys for already deleted data (with
+    // deletion_time set) when populating cash during plugin initialization.
+    // Need to fetch actual key and check it has valid version before using it.
+    if (!retrieve_key_type_and_data(key.get())) {
```

Comment:
@kamil-holubicki, in general, yes. I think I could drop actual data after verifying it. But I'm not sure this will change anything. The point is that it anyway will be cached later in Vault_keys_container::fetch_key(). If we drop data here during cache population it still will be fetched a few seconds later. Do you think it still makes sense?

Reply:
I don't know what is the use scenario. If it is as you say that ALL keys will be fetched anyway in a few seconds, the proposed solution is fine.